sstar/train.py

```diff
@@ -0,0 +1,796 @@
+
+    #drop_dynamic_cols indicate whether non-fixed size features should be dropped
+    if drop_dynamic_cols == True:
+        dynamic_cols = [col for col in train_df.columns if ('-ton' in col or col.startswith("pairwised_dist"))]
+        train_df.drop(dynamic_cols, axis=1, inplace = True, errors='ignore')
+
+    #train_df.to_csv(os.path.join(outputfolder, "features_final.csv"))
+    if do_training == True:
+        #start training
+        scikit_file = output_prefix + ".scikit.pickle"
+        statsmodels_file = output_prefix + ".statsmodels.pickle"
+
+        #call training functions
+        train_statsmodels(train_df, statsmodels_file)
+        train_scikit(train_df, scikit_file)
+
+    return train_df
+
+
+def _train_archie_folders(demo_model_file, nrep, nref, ntgt, ref_id, tgt_id, src_id, seq_len, mut_rate, rec_rate, thread, output_prefix, output_dir, scikit=True, statsmodels=True, drop_dynamic_cols=True, do_training = False):	
+    """
+    Description:
+        compute prdictions for files in all subdirectories of the path indicated by output_dirs
+
+    Arguments:
+        demo_model_file str: demographic model
+        nrep int: number of replicates
+        nref int: number of reference individuals
+        ntgt int: number of target individuals
+        ref_id str: name of reference population
+        tgt_id str: name of target population
+        src_id str: name os fource population
+        seq_len int: sequence length
+        mut_rate float: mutation rate
+        rec_rate float: recombination rate
+        threat int: number of threads
+        output_prefix str: string used to determine logistic regression model name
+        output_dir str: indicates folder with subdirectories containing files for prediction
+        scikit bool: if True, a model using scikit-learn is trained
+        statsmodels bool: if True, a model using statsmodels is trained
+        drop_dynamic_cols bool: if True, features of non-fixed size are removed
+        do_training: if True, models are trained; otherwise only statistics for training are computed
+
+    Returns:
+        train_df DataFrame: contains all windows of train data and corresponding information (statistics, label,...)
+    """
+
+
+    outputfolder = output_dir + "_features_df"
+
+    if not os.path.exists(outputfolder):
+        os.makedirs(outputfolder)    
+
+    #set filenames for individuals, reference and target
+    ref_ind_file = "new_sim" + "_nref" + str(nref) + "_ntgt" + str(ntgt) + ".ref.ind.list"
+    tgt_ind_file  = "new_sim" + "_nref" + str(nref) + "_ntgt" + str(ntgt) + ".tgt.ind.list"
+
+    create_ref_tgt_file(nref, ntgt, ref_ind_file, tgt_ind_file)
+
+    anc_allele_file = None
+
+    #set window length and stepsize
+    win_len = 50000
+    win_step = 50000
+   
+    #I think these parameters are NOT necessary for ArchIE - just retained for the signature of preprocess.process_data
+    match_bonus = 1
+    max_mismatch = 1
+    mismatch_penalty = 1
```

Comment:
These parameters are necessary for calculating the S* scores, which are also input features in the logistic regression.
```
match_bonus = 5000
max_mismatch = 5
mismatch_penalty = -10000
```